### ISSUE TITLE
Check for duplicate label names in remote read response

### ISSUE THREAD

**Issue by xinyu7git** [prometheus/prometheus#5731]
## Bug Report
**What did you do?**
Using remote_read and remote_write config, but Prometheus did not merge data from local storage and remote storage , i get two same points. so many function can't be used.

![image](https://user-images.githubusercontent.com/3701997/60559780-23973e80-9d80-11e9-86df-98072060bdd2.png)

**What did you expect to see?**
Prometheus should merge same points to one point.

**What did you see instead? Under which circumstances?**

**Environment**
   Start Command: /data/prometheus/prometheus --config.file=/data/prometheus/prometheus.yml --storage.tsdb.retention.time=2h --storage.tsdb.retention.size=2MB --log.level=debug

* System information:

	insert output of `uname -srm` here

         Linux 5.1.5-1.el7.elrepo.x86_64 x86_64
* Prometheus version:

	insert output of `prometheus --version` here
        `[root@10-0-0-14 prometheus]# ./prometheus --version
          prometheus, version 2.10.0 (branch: HEAD, revision: d20e84d0fb64aff2f62a977adc8cfb656da4e286)
          build user:       root@a49185acd9b0
          build date:       20190525-12:28:13
          go version:       go1.12.5`

* Alertmanager version:

	insert output of `alertmanager --version` here (if relevant to the issue)

* Prometheus configuration file:
```
insert configuration here
# my global config
global:
  scrape_interval:     15s # Set the scrape interval to every 15 seconds. Default is every 1 minute.
  evaluation_interval: 15s # Evaluate rules every 15 seconds. The default is every 1 minute.
  # scrape_timeout is set to the global default (10s).

# Alertmanager configuration
alerting:
  alertmanagers:
  - static_configs:
    - targets:
      # - alertmanager:9093

# Load rules once and periodically evaluate them according to the global 'evaluation_interval'.
rule_files:
  # - "first_rules.yml"
  # - "second_rules.yml"

# A scrape configuration containing exactly one endpoint to scrape:
# Here it's Prometheus itself.
scrape_configs:
  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
  - job_name: 'prometheus'

    # metrics_path defaults to '/metrics'
    # scheme defaults to 'http'.

    static_configs:
    - targets: ['10.0.0.14:9090']

  - job_name: 'node_exporter'
    scrape_interval: 10s
    static_configs:
    - targets: ['localhost:9100']

  - job_name: 'shanks-test'
    static_configs:
    - targets: ['10.0.0.14:9273']

remote_write:
  - url: "http://10.0.3.2:8086/api/v1/prom/write?db=prometheus"
remote_read:
  - url: "http://10.0.3.2:8086/api/v1/prom/read?db=prometheus"
    read_recent: true
```

* Alertmanager configuration file:
```
insert configuration here (if relevant to the issue)
```

* Logs:
```
insert Prometheus and Alertmanager logs relevant to the issue here
```

**Comment by xinyu7git** [comment]
When i use some function like avg_over_time, i get an error : 

`{
"status": "error",
"errorType": "execution",
"error": "vector cannot contain metrics with the same labelset"
} `

**Comment by xinyu7git** [comment]
i find the reason. ignore.

**Comment by brian-brazil** [comment]
What was the reason? This output shouldn't be possible.

**Comment by xinyu7git** [comment]
> What was the reason? This output shouldn't be possible.

In our elf-developed distributed cluster based on the influxdb oss version, there are some errors in the read and write logic of prometheus. Prometheus will take the label of __name__ when writing data to influxdb, and we used some codes in remote_storage_adapter when reading data from influxdb, it will automatically add a label of _name__, so there are two __name__ in the data returned to prometheus.

the prometheus remote_storage_adapter receive data from prometheus ,and then remove __name__ label and then write points to influxdb, but the influxdb(1.7.6) servicePromWrite API didnot remove the label , so points in influxdb have __name__ label(we didnot modify this part).  Influxdb servicePromRead API didnot use same logic like prometheus remote_storage_adapter, so we have encountered this error.

e.... same coes:
`func tagsToLabelPairs(name string, tags map[string]string) []prompb.Label {
        pairs := make([]prompb.Label, 0, len(tags))
        for k, v := range tags {
                if v == "" {
                        // If we select metrics with different sets of labels names,
                        // InfluxDB returns *all* possible tag names on all returned
                        // series, with empty tag values on series where they don't
                        // apply. In Prometheus, an empty label value is equivalent
                        // to a non-existent label, so we just skip empty ones here
                        // to make the result correct.
                        continue
                }
                pairs = append(pairs, prompb.Label{
                        Name:  k,
                        Value: v,
                })
        }
        pairs = append(pairs, prompb.Label{
                Name:  model.MetricNameLabel,
                Value: name,
        })
        return pairs
}`


The above code is the code in prometheus remote_storage_adapter example（influxdb/client.go)  . The tag of __name__ is added to the code. If you add the tag of __name__ repeatedly, you will have the same problem.
`func tagsToLabelPairs(name string, tags map[string]string) []prompb.Label {
        pairs := make([]prompb.Label, 0, len(tags))
        for k, v := range tags {
                if v == "" {
                        // If we select metrics with different sets of labels names,
                        // InfluxDB returns *all* possible tag names on all returned
                        // series, with empty tag values on series where they don't
                        // apply. In Prometheus, an empty label value is equivalent
                        // to a non-existent label, so we just skip empty ones here
                        // to make the result correct.
                        continue
                }
                pairs = append(pairs, prompb.Label{
                        Name:  k,
                        Value: v,
                })
        }
        pairs = append(pairs, prompb.Label{
                Name:  model.MetricNameLabel,
                Value: name,
        })
        pairs = append(pairs, prompb.Label{
                Name:  model.MetricNameLabel,
                Value: name,
        })
        return pairs
}`

I don't know if I have made it clear?

**Comment by brian-brazil** [comment]
Okay, so the problem is that we're not checking for duplicate label names in a remote read response.

**Comment by vmartsynovskyy** [comment]
I am working on this. I think I'm close to a solution, looks like there's a bug in `func (c *mergeSeriesSet) Next() bool`. I'll update with a PR once I've figured out a solution.

**Comment by cstyan** [comment]
fixed in #5829
